Introduce bruteforce settings

rullzer opened the pull request:
This adds an app with adds an admin settings. Lets the admin insert ip ranges (netmask) to exclude for throttling.

TODO:
- [x] Ignore ip ranges in throttler
- [x] Add JS tests
- [x] FIX CSS

@LukasReschke as discussed.

LukasReschke commented:
Let's add this to https://github.com/nextcloud/server/blob/9fff4b017d0ab6196f182ba2bb2c4fb00187a11b/.drone.yml#L22-L34 as well :)

rullzer committed:
Initial commit of the bruteforcesettings app

This adds the bruteforce settings app that allows to configure (for now)
subnets that are to be ignored when doing brute force analysis. This can
for example be the LAN since we trust people from there.

* Add app
* Add php tests
* Add js tests

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

rullzer committed:
Adds security section to the admin page

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

rullzer committed:
Respect bruteforce settings in the Throttler

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

rullzer commented:
@LukasReschke please have a look

rullzer committed:
App is always enabled

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

rullzer commented:
@MorrisJobke could you have a look at the crappy CSS? For some reason it wants text in order to show the trash icons

1. Go to admin page
2. Go to secrutiy tab
3. Add a subnet thingy
4. See bad icons
5. fix it 😉

MorrisJobke commented:
I will have a look once I'm in the office :wink:

MorrisJobke commented:
@rullzer The security page is empty 😢 

![bildschirmfoto 2016-11-18 um 21 25 12](https://cloud.githubusercontent.com/assets/245432/20445490/89e59ea4-add5-11e6-9f79-418ed5e84f3c.png)

rullzer commented:
Enable the app

MorrisJobke commented:
Ah ... 🙈

rullzer committed:
Update App Manager test

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

icewind1991 commented:
@rullzer any update here?

tx7 commented:
Hello all,

Any ETA on this?

MorrisJobke committed:
Fixed layout of bruteforcesettings

Signed-off-by: Morris Jobke <hey@morrisjobke.de>

MorrisJobke commented:
I rebased this, fixed the conflicts, improved the settings layout and it now looks like this:


<img width="852" alt="bildschirmfoto 2017-03-13 um 19 43 48" src="https://cloud.githubusercontent.com/assets/245432/23882348/962aff6a-0825-11e7-988a-cc741caf25de.png">


hovered:

<img width="729" alt="bildschirmfoto 2017-03-13 um 19 43 55" src="https://cloud.githubusercontent.com/assets/245432/23882352/9bc58c42-0825-11e7-98af-c4c22eba1c54.png">

MorrisJobke commented:
@nickvergessen @lukasreschke Ready for review \o/

icewind1991 commented:
@rullzer any reason to have this app in the server repo?

MorrisJobke commented:
> @rullzer any reason to have this app in the server repo?

Because it's force enabled for now.

rullzer commented:
@icewind1991 uhm. That was some time ago when I wrote this. But I would be fine with ripping it out... makes updating way easier ;)

icewind1991 commented:
@MorrisJobke we can still force enable apps that are in a seperate repo by bundling them for a release if needed.

rullzer commented:
Let me rip this apart soonish.

rullzer committed:
Fix app

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

rullzer committed:
Move app to own repo

https://github.com/nextcloud/bruteforcesettings

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

rullzer commented:
App in https://github.com/nextcloud/bruteforcesettings

rullzer committed:
Fix tests

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

rullzer commented:
Ok lets get this in!

MorrisJobke commented on tests/karma.config.js:
```diff
@@ -118,6 +118,19 @@ module.exports = function(config) {
+				],
+				testFiles: [
+					'apps/bruteforcesettings/tests/js/IPWhitelistSpec.js'
+				]
```
Shouldn't this be better executed in the apps repo? Otherwise the PR there aren't checked with the tests itself.

rullzer replied:
Yes it should.

MorrisJobke reviewed (approved):
Beside the one comment this is fine 👍

ChristophWurst commented on core/shipped.json:
```diff
@@ -2,6 +2,7 @@
   "shippedApps": [
     "activity",
     "admin_audit",
+    "bruteforcesettings",
```
@rullzer what's that used for? I don't see an app `bruteforcesettings`

rullzer replied:
Well it is just the list of shipped apps. I'm fine with removing that as well... if we decide to just ship it from the appstore...

ChristophWurst replied:
Oh, didn't see your comment https://github.com/nextcloud/server/pull/2095#issuecomment-291008958. Now it makes sense!
